### PR TITLE
PLT-6165 db sync comparison spent info

### PR DESCRIPTION
**Pull request by brendanrbrown**
Compares the output from `EventAtQuery` for the `SpentInfoEvent` SQLite indexer, for blocks in a few different eras.

* Does that with golden tests using the framework of PLT-7513 #282
* Adds a `DbSyncComparisonConfig` type and uses it in test builders.

As in #282, the tests are actually ignored until we have access to snapshot data in the CI on which to run the marconi queries.

I confirmed manually the golden files successfully decode via `FromJSON` to `SpentInfoEvent`.

Pre-submit checklist:
- Branch
    - [x] Tests are provided (if possible)
    - [x] Commit sequence broadly makes sense and have useful messages
    - [ ] Important changes are reflected in changelog.d of the affected packages
    - [x] Relevant tickets are mentioned in commit messages
- PR
    - [ ] (For external contributions) Corresponding issue exists and is linked in the description
    - [x] Targeting main unless this is a cherry-pick backport
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [ ] If relevant, reference the ADR in the PR and reference the PR in the ADR
    - [x] Reviewer requested

### CODE REVIEW

**Review by berewt — APPROVED** [review]
LGTM.

We definitely need to add long running tests to tests more slots/block though. We were talking about it with @ana-pantilie yesterday: have a few test for interesting blocks in CI and a nightly that tests all the blocks of some snapshots.

File: marconi-cardano-indexers/test/Spec/Marconi/Cardano/DbSyncComparison.hs [hunk 1]
```diff
@@ -13,37 +16,75 @@ tests =
...
+    , mkBlockInfoQueryBySlotNoTest $ blockInfoConfig Mainnet Alonzo1 43372792
+    , mkBlockInfoQueryBySlotNoTest $ blockInfoConfig Mainnet Alonzo2 43372972
+    , mkBlockInfoQueryBySlotNoTest $ blockInfoConfig Mainnet Alonzo2 45748924
+    , mkBlockInfoQueryBySlotNoTest $ blockInfoConfig Mainnet Alonzo2 48124747
+    , mkBlockInfoQueryBySlotNoTest $ blockInfoConfig Mainnet Babbage1 72316896
+    , mkBlockInfoQueryBySlotNoTest $ blockInfoConfig Mainnet Babbage1 77064584
+    , mkBlockInfoQueryBySlotNoTest $ blockInfoConfig Mainnet Babbage1 77068768
+    , mkBlockInfoQueryBySlotNoTest $ blockInfoConfig Mainnet Babbage2 84844885
+    , mkBlockInfoQueryBySlotNoTest $ blockInfoConfig Mainnet Babbage2 87198551
+    , mkBlockInfoQueryBySlotNoTest $ blockInfoConfig Mainnet Babbage2 89596758
```
Comment (ana-pantilie) [inline marconi-cardano-indexers/test/Spec/Marconi/Cardano/DbSyncComparison.hs:52]: My reasoning behind not adding a `blockInfoConfig` was that I'd like the test name to be easily modifiable for each test in particular. This is very useful if I want to just run one test based on its name. Unless you know a better way to do this, I wouldn't make this modification here.

Comment (brendanrbrown) [reply marconi-cardano-indexers/test/Spec/Marconi/Cardano/DbSyncComparison.hs]: Thanks for this. I made test name a parameter again, which I agree is a good idea.

**Comment by brendanrbrown** [comment]
> We definitely need to add long running tests to tests more slots/block though. 

Yeah good point. For this PR, I did add a few more tests so that we have at least one for each era --- at least for the eras for which I have db-sync data locally.

**Pull request merged**